### PR TITLE
all: try out mockito rule

### PR DESCRIPTION
**Pull request by carl-mastrangelo**

### CODE REVIEW

File: core/src/test/java/io/grpc/ForwardingChannelBuilderTest.java [hunk 1]
```diff
@@ -33,6 +33,8 @@
  */
 @RunWith(JUnit4.class)
 public class ForwardingChannelBuilderTest {
+
+
```
Comment (dapengzhang0) [inline core/src/test/java/io/grpc/ForwardingChannelBuilderTest.java]: Revert this

File: core/src/test/java/io/grpc/SynchronizationContextTest.java [hunk 2]
```diff
@@ -67,9 +71,6 @@ public void uncaughtException(Thread t, Throwable e) {
   @Mock
   private Runnable task3;
 
-  @Before public void setUp() {
-    MockitoAnnotations.initMocks(this);
-  }
```
Comment (dapengzhang0) [inline core/src/test/java/io/grpc/SynchronizationContextTest.java]: nit: remove one more line

File: core/src/test/java/io/grpc/ForwardingChannelBuilderTest.java (hunk 1 above)
Comment (carl-mastrangelo) [reply core/src/test/java/io/grpc/ForwardingChannelBuilderTest.java]: Done.

File: core/src/test/java/io/grpc/SynchronizationContextTest.java (hunk 2 above)
Comment (carl-mastrangelo) [reply core/src/test/java/io/grpc/SynchronizationContextTest.java]: Done.

**Comment by dapengzhang0** [comment]
May sure Windows is not more flaky as in #5219

**Comment by dapengzhang0** [comment]
@carl-mastrangelo You might need update examples/pom.xml and examples/gauth-example/build.gradle

**Comment by carl-mastrangelo** [comment]
@dapengzhang0 ping

**Pull request merged**